WooCommerce/Classes/ViewRelated/Products/Edit Product/Inventory Settings/ProductInventorySettingsViewController.swift

```diff
@@ -201,6 +198,36 @@ extension ProductInventorySettingsViewController {
 private extension ProductInventorySettingsViewController {
     func handleSKUChange(_ sku: String?) {
         self.sku = sku
+
+        // If the sku is identical to the old one, is always valid.
+        guard sku != product.sku else {
+            skuIsValid = true
+            hideError()
+            self.getSkuCell()?.textFieldBecomeFirstResponder()
+            return
+        }
+
+        // Throttled API call
+        let siteID = self.siteID
+        throttler.throttle {
+            DispatchQueue.main.async {
+                let action = ProductAction.validateProductSKU(sku, siteID: siteID) { [weak self] isValid in
+                    guard let self = self else {
+                        return
+                    }
+                    self.skuIsValid = isValid
+                    if isValid {
+                        self.hideError()
```

Comment:
if the validation passed, maybe we don't have to prompt the keyboard for the SKU field (`getSkuCell()?.textFieldBecomeFirstResponder()`)? maybe this could be a guard statement. for example, right now if I switch from the SKU field to the Stock > Quantity field, the input got switched back to the SKU field shortly after while the SKU is valid

Reply:
I understand what you mean. Yes, in this case, it makes sense that the text field becomes the first responder only if the SKU is not valid. 👍